lang/cs/Org.Apache.REEF.Network/NetworkService/WritableNsConnection.cs

```diff
@@ -0,0 +1,138 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net;
+using System.Net.Sockets;
+using System.Runtime.Remoting;
+using Org.Apache.REEF.Common.Io;
+using Org.Apache.REEF.Tang.Exceptions;
+using Org.Apache.REEF.Utilities.Logging;
+using Org.Apache.REEF.Wake;
+using Org.Apache.REEF.Wake.Remote;
+
+namespace Org.Apache.REEF.Network.NetworkService
+{
+    /// <summary>
+    /// Represents a connection between two hosts using the Writable NetworkService.
+    /// </summary>
+    [Obsolete("Need to remove Iwritable and use IstreamingCodec. Please see Jira REEF-295 ", false)]
+    public class WritableNsConnection<T> : IConnection<T> where T : IWritable
+    {
+        private static readonly Logger Logger = Logger.GetLogger(typeof (WritableNsConnection<T>));
+
+        private readonly IIdentifier _sourceId;
+        private readonly IIdentifier _destId;
+        private readonly INameClient _nameClient;
+        private readonly IRemoteManager<WritableNsMessage<T>> _remoteManager;
+        private readonly Dictionary<IIdentifier, IConnection<T>> _connectionMap;
+        private IObserver<WritableNsMessage<T>> _remoteSender;
+
+        /// <summary>
+        /// Creates a new NsConnection between two hosts.
+        /// </summary>
+        /// <param name="sourceId">The identifier of the sender</param>
+        /// <param name="destId">The identifier of the receiver</param>
+        /// <param name="nameClient">The NameClient used for naming lookup</param>
+        /// <param name="remoteManager">The remote manager used for network communication</param>
+        /// <param name="connectionMap">A cache of opened connections.  Will remove itself from
+        /// the cache when the NsConnection is disposed.</param>
+        public WritableNsConnection(
+            IIdentifier sourceId,
+            IIdentifier destId,
+            INameClient nameClient,
+            IRemoteManager<WritableNsMessage<T>> remoteManager,
+            Dictionary<IIdentifier, IConnection<T>> connectionMap)
+        {
+            _sourceId = sourceId;
+            _destId = destId;
+            _nameClient = nameClient;
+            _remoteManager = remoteManager;
+            _connectionMap = connectionMap;
+        }
+
+        /// <summary>
+        /// Opens the connection to the remote host.
+        /// </summary>
+        public void Open()
+        {
+            string destStr = _destId.ToString();
+            Logger.Log(Level.Verbose, "Network service opening connection to {0}...", destStr);
+
+            IPEndPoint destAddr = _nameClient.Lookup(_destId.ToString());
+            if (destAddr == null)
+            {
+                throw new RemotingException("Cannot register Identifier with NameService");
```

Comment:
the error is that the identifier cannot be found, correct? The message indicates otherwise.